tests/test_cardinality_constraint_discrete.py

```diff
@@ -0,0 +1,53 @@
+"""Test for the discrete cardinality constraint."""
```

Comment:
why this file separation?

if the test files get a little bit too long I'd prefer if we make another foler substructure instead of having several new files in the `test` folder
eg
test/
test/serialization
...
test/constraints
test/constraints/test_cardinality.py

etc

Reply:
Thanks! A subfolder test/constraints/ has been created, containing all constraints-related test files:
- test_constraints_discrete.py
- test_constraints_continuous.py
- test_cardinality_constraint_discrete.py
- test_cardinality_constraint_continuous.py